nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-services/src/main/java/org/apache/nifi/mongodb/MongoDBLookupService.java

```diff
@@ -52,68 +54,125 @@
     "The query is limited to the first result (findOne in the Mongo documentation). If no \"Lookup Value Field\" is specified " +
     "then the entire MongoDB result document minus the _id field will be returned as a record."
 )
-public class MongoDBLookupService extends MongoDBControllerService implements LookupService<Object> {
+public class MongoDBLookupService extends SchemaRegistryService implements LookupService<Object> {
+    public static final PropertyDescriptor CONTROLLER_SERVICE = new PropertyDescriptor.Builder()
```

Comment:
AFAICT this property is never added to the list of supported property descriptors, so I couldn't set it on the UI which causes an NPE when lookup() is called. Seems odd that for a required property that is not supported, setting it (in tests) would not complain. I haven't run the integration tests yet, just put the NARs into a live NiFi to try it out.